ddtrace/contrib/jinja2/patch.py

```diff
@@ -0,0 +1,131 @@
+import jinja2
+from wrapt import wrap_function_wrapper as _w
+
+from ddtrace import config
+
+from ...ext import http
+from ...utils.formats import asbool, get_env
+from ...pin import Pin
+from ...utils.wrappers import unwrap as _u
+from .constants import DEFAULT_SERVICE, DEFAULT_TEMPLATE_NAME
+
+
+# default settings
+config._add('jinja2',{
+    'service_name': get_env('jinja2', 'service_name', DEFAULT_SERVICE),
+    'inherit_service': asbool(get_env('jinja2', 'inherit_service', True)),
+})
+
+
+def patch():
+    if getattr(jinja2, "__datadog_patch", False):
+        # already patched
+        return
+    setattr(jinja2, "__datadog_patch", True)
+    Pin(
+        service=config.jinja2['service_name'],
+        app='jinja2',
+        app_type=http.TEMPLATE,
+        _config=config.jinja2,
+    ).onto(jinja2.environment.Environment)
+    _w(jinja2, "environment.Template.render", _wrap_render)
+    _w(jinja2, "environment.Template.generate", _wrap_generate)
+    _w(jinja2, "environment.Environment.compile", _wrap_compile)
+    _w(jinja2, "environment.Environment._load_template", _wrap_load_template)
+
+
+def unpatch():
+    if not getattr(jinja2, "__datadog_patch", False):
+        return
+    setattr(jinja2, "__datadog_patch", False)
+    _u(jinja2.Template, "render")
+    _u(jinja2.Template, "generate")
+    _u(jinja2.Environment, "compile")
+    _u(jinja2.Environment, "_load_template")
+
+
+def _get_service_name(environment, span):
```

Comment:
I think by default here, given the nature of jinja, we will want the service to be inherited by default. 

We should let the tracer handle the parenting of the service name unless a service name is provided. We should default to having `cfg['service_name']=None` and then we no longer need this function as we can just use `cfg['service_name']`.

This avoids dealing with the internals of the span and gives us the same functionality.

Reply:
do you confirm I should have `span.service = pin.service` instead of this function ?

Reply:
ok, it's better to add pin.service to `trace()`